client/src/elements/schema-editor/schema-editor-geojson-point.js

```diff
@@ -53,335 +39,251 @@ export class SchemaEditorGeojsonPoint {
-              return allResults;
-            }, {});
-        },
-        filterData: (value, records) => {
-          return records; // do not filter the results
-        },
-        minLength: 3,
-        autoCollapse: true,
-        marker: false,
-        textPlaceholder: ""
-      });
-
-      this.geocoder.on("search:locationfound", selectedSearchResult => {
-        this.data.geometry = {
-          type: "Point",
-          coordinates: [
-            selectedSearchResult.latlng[1],
-            selectedSearchResult.latlng[0]
-          ]
-        };
+    // Add marker if existing point is displayed
+    if (
+      this.data.geometry &&
+      this.data.geometry.coordinates &&
+      this.data.geometry.coordinates.length >= 2 &&
+      !this.marker
+    ) {
+      this.updateMarker();
+    }
 
-        // try to find the selected result by latlng
-        let selected = currentSearchResults.filter(result => {
-          return (
-            result.geometry.coordinates[0] === selectedSearchResult.latlng[1] && // yes it needs to be 1st vs. 2nd
-            result.geometry.coordinates[1] === selectedSearchResult.latlng[0]
-          ); // lat lng / lng lat mixup is a major fuckup
-        })[0];
+    if (
+      this.options.bbox === "manual" &&
+      this.data.bbox &&
+      this.data.bbox.length === 4
+    ) {
+      this.map.fitBounds(this.data.bbox, { duration: 0 });
+    }
 
-        if (selected && selected.properties) {
-          try {
-            if (
-              selected.properties.components[
-                selected.properties.components._type
-              ]
-            ) {
-              this.data.properties.label =
-                selected.properties.components[
-                  selected.properties.components._type
-                ];
-            } else {
-              throw new Error("property not available");
-            }
-          } catch (e) {
-            this.data.properties.label = selected.properties.formatted;
-          }
-        }
-        if (this.change) {
-          this.change();
-        }
-        this.updateMarker();
+    if (this.options.bbox === "manual") {
+      this.map.on("mouseout", event => {
+        this.updateBBox();
       });
     }
 
-    this.zoomControl = window.Leaflet.control.zoom({
-      position: "topleft"
-    });
-
-    // allows to add a pin by clicking the map only if there is no pin yet
-    this.map.on("click", e => {
-      if (!this.pin) {
-        this.data.geometry.coordinates = [e.latlng.lng, e.latlng.lat];
+    // Add marker to the clicked area
+    this.map.on("click", event => {
+      if (!this.marker) {
+        this.data.geometry.coordinates = [event.lngLat.lng, event.lngLat.lat];
         this.updateMarker();
-        if (this.change) {
-          this.change();
+        if (this.options.bbox === "manual") {
+          this.updateBBox();
         }
       }
     });
 
-    const layerConfig = schemaEditorConfig.geojson.layer;
```

Comment:
If the support for a tilelayer config gets removed, I would consider this a breaking change. If it is not too crazy to support the new config with a style url and the old one with a tile layer config (mapbox-gl can still display normal raster tile layers right?) then I would prefer to keep the support for this as it makes it easier to update the editor for all instances we are running.